packages/documentation/src/pages/architecture/unit-tests.mdx

```diff
@@ -0,0 +1,120 @@
+For e2e tests:
+- It's possible to set an infinite pause (`.pause()`) in the test code and then see what's happening at a given point in the browser where the test is running. It is also possible to use the Chrome devtools and inspect what's going on (this is only possible when the browser is Chrome and not Electron, so you would need to disable Electron is `config/nightwatch.js`).
+
+
+#### When we write them
+Write tests as you go! New functionality added in a PR should ideally be covered by unit and e2e tests, where applicable.
+
+
+### Libraries
+
+1. Test environment
+- **mocha**: testing framework
+
+2. Test structure
+- **mocha**: testing framework
+
+3. Assertion functions
+- **chai**: assertion library, can be used with an JS testing framework
+- **chai-as-promised**: includes assertions for promises
+
+4. Test utilities/helpers
+- **enzyme**: testing utility for React, use in place of React test-utils
+- **enzyme-adapter-react-16**
+- **fs**: makes file system operations easier
+- **lodash**: library with utility functions for commonly used tasks, using functional programming
+- **moment**: library for parsing, manipulating, and displaying dates and times in JS
+- **path**: utilities for working with file and directory paths
+- **react-dom**: used for findDOMNode
+- **react-dom/test-utils**: test utilities for React components, being replaced by Enzyme
+- **skin-deep**: test helpers using React's shallowRender (Deprecated in favor of Enzyme)
+
+5. Display and watch test results
+- **mocha**: testing framework
+
+6. Mocks, spies, stubs
+- **sinon**: test spies, stubs, and mocks
+
+7. Browser environment
+- **chromedriver**: a server that links the browser (Chrome) to the test driver (Selenium), enabling you to execute actions in the browser through a test
+- **jsdom**: JS headless browser that creates a realistic testing environment with a DOM to which elements can be mounted
+- **nightwatch**: Node.js framework for e2e tests that runs on a selenium server
+- **saucelabs**: automated cross-browser testing, allows you to run tests on a server
+- **selenium-server**: automates browsers
+
+8. Linting
+- **eslint**: for linting purposes
+- **eslint-config-airbnb**: Airbnb's lint rules
+- **eslint-plugin-jsx-a11y**: catches potential accessibility issues
+- **eslint-plugin-no-unsafe-innerhtml**: catches potential security vulnerabilities
+- **eslint-plugin-react**: React specific linting rules
+- **eslint-plugin-scanjs-rules**: static analysis of JS code
+- **sass-lint**: linter for sass and scss
+
+9. Visual regression testing
+- **puppeteer**: Automates browser testing
+- **resemble**: Image analysis
+
+10. Accessibility testing
+- **axe-core**: automated accessibility testing
+
+11. Original libraries
+- **jsonschema**: JSON Schema specification
+- **react**: JS front-end framework
+- **react-jsonschema-form**: react-jsonschema-form library
+- **react-test-renderer**: renders React components to pure JavaScript objects
+
+
+### Questions
```

Comment:
These questions are sort of arbitrary- we should try to organize it into the content above.